### ISSUE TITLE
Duplicate modulepreload links in head.

### ISSUE THREAD

**Issue by roelvanhintum** [nystudio107/craft-vite#16]
Some modulepreload link's are added twice (notice vendor in the case below) in specific manifest cases. My guess it's because of the 3 input scripts, with each slightly different imports in the js itself.

Output in html (domain redacted)
```html
<script type="module" src="/dist/assets/main.168516cf.js" crossorigin></script>
<link href="/dist/assets/Layout.5daf469e.js" rel="modulepreload" crossorigin>
<link href="/dist/assets/vendor.01ab196e.js" rel="modulepreload" crossorigin>
<link href="/dist/assets/vendor.01ab196e.js" rel="modulepreload" crossorigin>
<link href="/dist/assets/main.575548a8.css" rel="stylesheet" media="print" onload="this.media=&#039;all&#039;">
```

Used manifest.json
```json
{
  "src/js/main.js": {
    "file": "assets/main.168516cf.js",
    "src": "src/js/main.js",
    "isEntry": true,
    "imports": [
      "_Layout.5daf469e.js",
      "_vendor.01ab196e.js"
    ],
    "css": [
      "assets/main.575548a8.css"
    ],
    "assets": [
      "assets/Font-Regular.746d0c22.woff2",
      "assets/Font-Regular.1f967c79.woff",
      "assets/Font-Bold.6c627e8a.woff2",
      "assets/Font-Bold.5d3d5626.woff",
      "assets/gradient-category-cta.0cafa9fc.png",
      "assets/footer-bg.b5cbcf69.svg",
      "assets/gradient-bg-mob.7b9a948d.jpg",
      "assets/gradient-bg.d3559d31.jpg",
      "assets/home-footer-bg.3a09858a.png"
    ]
  },
  "src/js/footer.js": {
    "file": "assets/footer.1b22b364.js",
    "src": "src/js/footer.js",
    "isEntry": true,
    "css": [
      "assets/footer.f52685bf.css"
    ],
    "assets": [
      "assets/Font-Regular.746d0c22.woff2",
      "assets/Font-Regular.1f967c79.woff",
      "assets/Font-Bold.6c627e8a.woff2",
      "assets/Font-Bold.5d3d5626.woff"
    ]
  },
  "src/js/menu.js": {
    "file": "assets/menu.02571f38.js",
    "src": "src/js/menu.js",
    "isEntry": true,
    "imports": [
      "_Layout.5daf469e.js",
      "_vendor.01ab196e.js"
    ],
    "css": [
      "assets/menu.b2391e91.css"
    ],
    "assets": [
      "assets/Font-Regular.746d0c22.woff2",
      "assets/Font-Regular.1f967c79.woff",
      "assets/Font-Bold.6c627e8a.woff2",
      "assets/Font-Bold.5d3d5626.woff"
    ]
  },
  "_vendor.01ab196e.js": {
    "file": "assets/vendor.01ab196e.js"
  },
  "_Layout.5daf469e.js": {
    "file": "assets/Layout.5daf469e.js",
    "imports": [
      "_vendor.01ab196e.js"
    ]
  }
}
```

### Versions
- Plugin version: 1.0.16
- Craft version: Craft Pro 3.7.16

**Comment by khalwat** [comment]
I think it's probably not actually harming anything, browsers should handle this gracefully, but agree we should de-dup these and fix the issue!

**Comment by roelvanhintum** [comment]
@khalwat, you're right, everything works fine despite the duplicates.

**Comment by khalwat** [comment]
So my thought here is that instead of just pushing these into an array, we can push them with the path as a key, and so can eliminate any duplicates easily there.

**Comment by khalwat** [comment]
Addressed in: https://github.com/nystudio107/craft-plugin-vite/commit/2bd4b7812bcf2b47f13c5cfe4fde01146efab6e0

Released as Craft Vite 1.0.17 -> https://github.com/nystudio107/craft-vite/releases/tag/1.0.17
